### PR TITLE
typing & PEP 561

### LINKED ISSUE #52: Downstream type hints?
First of all, thanks for this fantastic library!

I'm currently working on using `cyclonedx-python-lib` to generate SBOMs within [`pip-audit`](https://github.com/trailofbits/pip-audit).

Our codebase is fully typechecked (via MyPy), and it looks like this library is too! But we can't currently consume your type hints, since the `cyclonedx` package doesn't include a `py.typed` file per [PEP 561](https://www.python.org/dev/peps/pep-0561/). 

Is there any objection to including that file, so that downstream users can typecheck their API use? If not, I can go ahead and create a PR -- it should be just a 2 or 3 line change.

### PR DESCRIPTION
**Pull request by jkowalleck**
* added [PEP 561](https://www.python.org/dev/peps/pep-0561/) support
* added type checker/ static code analysis to the CI/CT

closes  #52

checked and confirmed, that `poetry build` results contain the file `py.typed` in the `wheel` and in the `tar.gz` as expected

----

:construction:  TODO :construction: 
- [x]  fully type all the methods

### CODE REVIEW

File: cyclonedx/model/bom.py [hunk 1]
```diff
@@ -37,11 +37,11 @@ class Tool:
         See the CycloneDX Schema for toolType: https://cyclonedx.org/docs/1.3/#type_toolType
     """
 
-    def __init__(self, vendor: str, name: str, version: str, hashes: List[HashType] = []):
```
Comment (jkowalleck) [inline cyclonedx/model/bom.py:40]: the default `[]` list was mitable.
see https://blog.finxter.com/the-mutable-default-argument-in-python/

File: cyclonedx/model/component.py [hunk 2]
```diff
@@ -19,8 +19,10 @@
 
 from enum import Enum
 from os.path import exists
-from packageurl import PackageURL
-from typing import List
+from typing import List, Optional
+
+# See https://github.com/package-url/packageurl-python/issues/65
```
Comment (jkowalleck) [inline cyclonedx/model/component.py:24]: https://github.com/package-url/packageurl-python/issues/65

File: cyclonedx/utils/conda.py [hunk 3]
```diff
@@ -41,26 +41,23 @@ class CondaPackage(TypedDict):
...
-        package_data['md5_hash'] = None
-
-    if isinstance(package_data, dict):
-        return CondaPackage(**package_data)
+    if not isinstance(package_data, dict):
+        return None
 
-    return None
+    package_data.setdefault('md5_hash', None)
+    return CondaPackage(package_data)  # type: ignore # @FIXME write proper type safe dict at this point
```
Comment (jkowalleck) [inline cyclonedx/utils/conda.py:57]: I thought this was good enough
what do you think, @madpah ?

File: cyclonedx/model/bom.py (hunk 1 above)
Comment (jkowalleck) [reply cyclonedx/model/bom.py:40]: added `flake8-bugbear` to the project, which would have found such mistakes.

File: cyclonedx/utils/conda.py (hunk 3 above)
Comment (madpah) [reply cyclonedx/utils/conda.py:57]: Can't think of another way without more invasive changes for now. Happy enough if you are @jkowalleck!

**Pull request merged**